mlflow/llama_index/__init__.py

```
@@ -0,0 +1,338 @@
+        signature: A Model Signature object that describes the input and output Schema of the
+            model. The model signature can be inferred using `infer_signature` function
+            of `mlflow.models.signature`.
+        input_example: {{ input_example }}
+        pip_requirements: {{ pip_requirements }}
+        extra_pip_requirements: {{ extra_pip_requirements }}
+        conda_env: {{ conda_env }}
+        metadata: {{ metadata }}
+    """
+
+    _validate_engine_type(engine_type)
+    _validate_env_arguments(conda_env, pip_requirements, extra_pip_requirements)
+
+    path = os.path.abspath(path)
+    _validate_and_prepare_target_save_path(path)
+    code_dir_subpath = _validate_and_copy_code_paths(code_paths, path)
+
+    if signature is None and input_example is not None:
+        wrapped_model = create_engine_wrapper(index, engine_type, model_config)
+        signature = infer_signature_from_input_example(input_example, wrapped_model)
+    elif signature is False:
+        signature = None
+
+    if mlflow_model is None:
+        mlflow_model = Model()
+    if signature is not None:
+        mlflow_model.signature = signature
+    if input_example is not None:
+        _save_example(mlflow_model, input_example, path)
+    if metadata is not None:
+        mlflow_model.metadata = metadata
+
+    # NB: llama_index.core.Settings is a singleton that manages the storage/service context
+    # for a given llama_index application. Given it holds the required objects for most of
+    # the index's functionality, we look to serialize the entire object. For components of
+    # the object that are not serializable, we log a warning.
+    settings_path = os.path.join(path, _SETTINGS_DIRECTORY)
+    serialize_settings(settings_path)
+
+    _save_model(index, path)
+
+    if not model_config:
+        model_config = {"engine_type": engine_type}
+    elif isinstance(model_config, dict):
+        model_config.update({"engine_type": engine_type})
```

Shall we raise an exception when `model_config` is not dictionary?